### PR TITLE
Notification Badge: Created and converted test scripts to puppeteer 

### PR DESCRIPTION
**Pull request by jbrcna**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**
      I also added some reusable functions in e2e-utils.js 
      To  know more about convenience functions click [README](https://gist.github.com/jbrcna/516f8ef4831be9054859f5812a17d408#file-readme-md)
      I'm still learning things and I'm open to ideas :slightly_smiling_face:

**Related github/jira issue (required)**:
  ✓ Created puppeteer scripts for (Enhancement) 5344
  ✓ Added Badge Placement Tests
  ✓ Added Index Tests
PASS testcomponentsnotification-badgenotification-badge puppeteer spec js (6 126

**Steps necessary to review your pull request (required)**:

1. pull this branch
2. build and run the app
3. run npm run e2e:puppeteer notification-badge
4. It should pass.

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.
- [x] Helper Functions

### CODE REVIEW

**Review by EdwardCoyle — DISMISSED** [review]

**Comment by tmcconechy** [comment]
Pr is merged you can resume this now @jbrcna

**Comment by jbrcna** [comment]
> Pr is merged you can resume this now @jbrcna

Alright! Thanks! I'm currently working on the scripts for charts enhancement 2736 but will park it cause ticket#2736 has a lot of components.

**Comment by jbrcna** [comment]
Hello @tmcconechy , this is now ready for review  :slightly_smiling_face: . I hope I didn't get too many things mixed up this time. :sweat_smile:
image

File: test/components/notification-badge/notification-badge.puppeteer-spec.js [hunk 1]
```diff
@@ -0,0 +1,548 @@
...
+            expect(width).toBe(value);
+          } catch (err) {
+            hasFailed = true;
+          }
+          index += 1;
+        }
+        return hasFailed;
+      };
+
+      const checkiconSize = async (style, value) => {
```
Comment (tmcconechy) [inline test/components/notification-badge/notification-badge.puppeteer-spec.js:445]: With these this is a case where you could make a function at the top of the test and reuse it within the test...

Also make sure that none are duplicated as it just adds to the time it takes to run it. IE do we need to check the size on all examples or just one should be fine?

**Review by tmcconechy — COMMENTED** [review]
Some suggestions. this is looking pretty nice

File: test/components/notification-badge/notification-badge.puppeteer-spec.js [hunk 2]
```diff
@@ -0,0 +1,548 @@
...
+      isFailed.push(await checkInnerHTMLValue('#notification-badge-3 > .notification-badge-container', 'notification-dot-upper-right'));
+      isFailed.push(await checkInnerHTMLValue('#notification-badge-4 > .notification-badge-container', 'notification-dot-upper-right'));
+      isFailed.push(await checkInnerHTMLValue('#notification-badge-5 > .notification-badge-container', 'notification-dot-upper-right'));
+      isFailed.push(await checkInnerHTMLValue('#notification-badge-6 > .notification-badge-container', 'notification-dot-upper-right'));
+      expect(isFailed).not.toContain(true);
+    });
+  });
+
+  describe('Buttons Tests', () => {
+    /*
```
[the remark this answers is not in the document]
Comment (jbrcna) [reply test/components/notification-badge/notification-badge.puppeteer-spec.js:181]: this is now removed :slightly_smiling_face:. sorry for the delay I was on leave yesterday.

File: test/components/notification-badge/notification-badge.puppeteer-spec.js (hunk 1 above)
Comment (jbrcna) [reply test/components/notification-badge/notification-badge.puppeteer-spec.js:445]: thanks for the tip @tmcconechy . I removed all duplicates and agree that one should be fine.

Comment (jbrcna) [reply test/components/notification-badge/notification-badge.puppeteer-spec.js:445]: thanks for the tip @tmcconechy . I removed all duplicates and agree that one should be fine.

**Comment by jbrcna** [comment]
image

**Review by tmcconechy — DISMISSED** [review]

File: test/components/notification-badge/notification-badge.puppeteer-spec.js [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+      await expect(page.title()).resolves.toMatch('IDS Enterprise');
+    });
+
+    it('should check the test page with Axe', async () => {
+      await page.setBypassCSP(true);
+      await page.goto(url, { waitUntil: ['domcontentloaded', 'networkidle0'] });
+      await expect(page).toPassAxeTests({ disabledRules: ['meta-viewport'] });
+    });
+
+    it('should be accessible', async () => {
```
Comment (tmcconechy) [inline test/components/notification-badge/notification-badge.puppeteer-spec.js:20]:
```suggestion
    it('should pass accessibility checks', async () => {
```

File: test/components/notification-badge/notification-badge.puppeteer-spec.js [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
```
Comment (tmcconechy) [inline test/components/notification-badge/notification-badge.puppeteer-spec.js:142]:
```suggestion
    it('should pass accessibility checks', async () => {
```

**Review by tmcconechy — COMMENTED** [review]
We used this name on another PR

**Comment by tmcconechy** [comment]
@jbrcna please update this branch with main. Also you should be able to use branches that way i could update things for you fx

```
git checkout main
git pull
git checkout -b my-branch-for-tests
// do work
git commit -m
git push
// make pr from the branch
```

**Comment by jbrcna** [comment]
> @jbrcna please update this branch with main. Also you should be able to use branches that way i could update things for you fx
> 
> ```
> git checkout main
> git pull
> git checkout -b my-branch-for-tests
> // do work
> git commit -m
> git push
> // make pr from the branch
> ```

This is one of the first scripts I created, which is why its branch is not on main. I don't have access to branches during that time. Rather than creating a new PR, I just kept working on this one. However, all subsequent scripts were already using branches in main.

I appreciate your patience @tmcconechy . Thank you for assisting me in understanding git/github and everything else.

**Review by tmcconechy — APPROVED** [review]

**Review by tjamesallen15 — APPROVED** [review]

**Pull request merged**